### ISSUE TITLE
"hasClass" logic in toggleClass function

### ISSUE THREAD

**Issue by DoraemonYu** [jstools/jqlite#42]
o(∩_∩)o Hello~

I use this light weight these days, and I found a small logic that seems need to be adjusted.
In the newest source code of "jqlite.js", the part of  the function of [ListDOM.prototype.toggleClass] as below:

```
if (add === undefined) {
    for (i = 0, len = this.length; i < len; i++) {
          if (this[i].classList.item(className)) {
              this[i].classList.remove(className);
          } else {
              this[i].classList.add(className);
          }
    }
}
```
this[i].classList.**item**(className)

When I call **toggleClass** function  many times,I found the result is not expected.
Therefore,I saw the function of **hasClass** function, and see the check logic is:
this[i].classList.**contains**(className)

So, I modify code use **contains** instead of **item**, then the result is correct now.

Thank you for your reading.

**Comment by BranoMatan** [comment]
"item" method gets an index (integer) as the input, not the class name (string probably). That is why the result was unexpected :)

https://developer.mozilla.org/en-US/docs/Web/API/Element/classList

**Comment by jgermade** [comment]
Thanks @DoraemonYu for the feedback about this issue.
And thanks @DoraemonYu and @BranoMatan for the tip for the solution.

I'm working in the patch and tests...

**Comment by jgermade** [comment]
I've finally refactored classList (and falback) handling. Also added support for function className.

Available on release [0.2.39](https://github.com/jstools/jqlite/releases/tag/v0.2.39)
